File: 3702-Longest-Subsequence-With-Non-Zero-Bitwise-XOR/Solution.py
class Solution:
    def longestSubsequence(self, nums: list[int]) -> int:
        n = len(nums)
        xor = 0
        has_non_zero = False

        for num in nums:
            xor ^= num
            if num != 0:
                has_non_zero = True

        if xor != 0:
            return n
        return n - 1 if has_non_zero else 0

File: 3702-Longest-Subsequence-With-Non-Zero-Bitwise-XOR/Solution.rs
impl Solution {
    pub fn longest_subsequence(nums: Vec<i32>) -> i32 {
        let n = nums.len() as i32;
        let mut xor = 0;
        let mut has_non_zero = false;

        for num in nums {
            xor ^= num;
            if num != 0 {
                has_non_zero = true;
            }
        }

        if xor != 0 {
            n
        } else if has_non_zero {
            n - 1
        } else {
            0
        }
    }
}
